relayer/src/services/relayer/ethereum.ts

```diff
@@ -128,64 +55,62 @@ export class EthereumRelayerService extends BaseRelayerService {
     return keccak256(encodedCall);
   }
 
-  async prepare(): Promise<boolean> {
-    return true;
-  }
-
-  async relayTxToL1(l1TransactionHash: TransactionHash): Promise<string | undefined> {
-    console.log("\nExecuting message on Ethereum L1.");
-
-    const callParameters = await this.parseEventDataFromTxHash(l1TransactionHash);
-
-    if (!callParameters) {
-      throw new Error(
-        "Couldn't find a matching event (RequestFilled | FillInvalidated) in the transaction logs.",
-      );
+  private async isRelayCompleted(
+    l1TransactionHash: TransactionHash,
+  ): Promise<TransactionHash | false> {
+    if (!this.callParameters) {
+      this.callParameters = await this.parseFillEventDataFromTxHash(l1TransactionHash);
```

Comment:
Same as in `arbitrum.ts`. Lets declare this variable locally in the scope of this function instead of the class scope.

Reply:
:heavy_check_mark: